contracts/ReputationMiningCycleRespond.sol

```diff
@@ -265,59 +265,45 @@ contract ReputationMiningCycleRespond is ReputationMiningCycleStorage, PatriciaT
     return afterInsertionAdjacentReputationSiblings;
   }
 
-  function checkOriginReputation(
+  function checkUserOriginReputation(
     uint256[27] memory u,
     bytes[5] memory b,
     bytes32[] memory agreeStateSiblings,
-    bytes32[] memory originReputationSiblings) internal 
+    bytes32[] memory userOriginReputationSiblings) internal 
   {
     ReputationLogEntry storage logEntry = reputationUpdateLog[u[U_LOG_ENTRY_NUMBER]];
     if (logEntry.amount > 0) {
       return;
     }
 
     bytes memory reputationKey = b[B_REPUTATION_KEY];
-    bytes memory originReputationKey = b[B_ORIGIN_REPUTATION_KEY];
+    bytes memory userOriginReputationKey = b[B_ORIGIN_REPUTATION_KEY];
 
     uint256 relativeUpdateNumber = getRelativeUpdateNumber(u, logEntry);
     uint256 nChildUpdates;
     (nChildUpdates, ) = getChildAndParentNUpdatesForLogEntry(u);
 
     // Skip origin reputation checks for anything but child reputation updates
     if (relativeUpdateNumber % (logEntry.nUpdates/2) < nChildUpdates) {
-      // Check the origin reputation key matches the colony, user address and skill id of the child skill
+      // Check the user origin reputation key matches the colony, user address and skill id of the log
       address colonyAddressOriginRep;
       address userAddressOriginRep;
       uint256 skillIdOriginRep;
       assembly {
-          colonyAddressOriginRep := mload(add(originReputationKey,20))
-          skillIdOriginRep := mload(add(originReputationKey, 52))
-          userAddressOriginRep := mload(add(originReputationKey,72))
+          colonyAddressOriginRep := mload(add(userOriginReputationKey,20))
+          skillIdOriginRep := mload(add(userOriginReputationKey, 52))
+          userAddressOriginRep := mload(add(userOriginReputationKey,72))
       }
 
-      address colonyAddressChildRep;
-      address userAddressChildRep;
-      assembly {
-          colonyAddressChildRep := mload(add(reputationKey,20))
-          userAddressChildRep := mload(add(reputationKey,72))
-      }
-
-      require(colonyAddressOriginRep == colonyAddressChildRep, "colony-reputation-mining-origin-colony-incorrect");
+      require(colonyAddressOriginRep == logEntry.colony, "colony-reputation-mining-origin-colony-incorrect");
       require(skillIdOriginRep == logEntry.skillId, "colony-reputation-mining-origin-skill-incorrect");
-
-      // For colony wide updates the reputation key userAddress is 0x0, otherwise ensure it the user address of the origin and child skills match
-      if (relativeUpdateNumber < nChildUpdates) {
-        // I don't think this require is possible to hit - I can't see how to get checkKeyLogEntry to pass and this to fail
-        require(userAddressChildRep == address(0x0), "colony-reputation-mining-colony-wide-update-user-nonzero");
```

Comment:
I am guessing I added this before you did the edge case of disputes over colony wide child reputations and so this bug was regressed :(

Reply:
I've definitely got some blood on my hands here! I remember futzing around with this code at the time (the comment about not being able to hit the require is definitely mine) and it didn't click then.